src/lib/routeGenerators.js

```diff
@@ -0,0 +1,85 @@
+
+const Artists = require('../models/artists')
+const Genres = require('../models/genres')
+const Albums = require('../models/albums')
+const artists = new Artists()
+const genres = new Genres()
+const albums = new Albums()
+
+function getModel (req, res, next) {
+  const model = req.params.model
+  switch (model) {
+    case 'artists':
+      req.model = artists
+      next()
+      return
+    case 'genres':
+      req.model = genres
+      next()
+      return
+    case 'albums':
+      req.model = albums
+      next()
+      return
+    default:
+      throw new Error('Invalid Model')
+  }
+}
+
+function handleGet (req, res, next) {
+  const input = req.params.id ? req.params.id : ''
+  req.model.read(input)
+    .then(results => {
+      const output = {
+        count: results.length,
+        data: results
+      }
+      res.status(200).json(output)
+    })
+    .catch(next)
+}
+
+function handleGetOneArtist (req, res, next) {
+  req.model.read(req.params.id)
+    .populate('albums')
```

Comment:
This function will grab a particular artist and return their info. If they have an album associated, it will show the artist's album information as well.
![image](https://user-images.githubusercontent.com/16282245/75481262-9ed4e280-5957-11ea-8c79-f2d27cae69e8.png)